tests/components/blackbird/test_media_player.py

```diff
@@ -171,172 +169,190 @@ def test_invalid_schemas(self):
                 PLATFORM_SCHEMA(value)
 
 
-class TestBlackbirdMediaPlayer(unittest.TestCase):
-    """Test the media_player module."""
-
-    def setUp(self):
-        """Set up the test case."""
-        self.blackbird = MockBlackbird()
-        self.hass = tests.common.get_test_home_assistant()
-        self.hass.start()
-        # Note, source dictionary is unsorted!
-        with mock.patch(
-            "homeassistant.components.blackbird.media_player.get_blackbird",
-            new=lambda *a: self.blackbird,
-        ):
-            setup_platform(
-                self.hass,
-                {
-                    "platform": "blackbird",
-                    "port": "/dev/ttyUSB0",
-                    "zones": {3: {"name": "Zone name"}},
-                    "sources": {
-                        1: {"name": "one"},
-                        3: {"name": "three"},
-                        2: {"name": "two"},
-                    },
+@pytest.fixture
+def blackbird():
+    """Fixture to mock Blackbird."""
+    yield MockBlackbird()
```

Comment:
My impression is that "yield" is used to allow for a setup and teardown, and I don't see any teardown happening here.  Can this jus the return? or is there something else this is doing.  Similar question below also.